bottles/backend/models/btrfssubvolume.py

```diff
@@ -0,0 +1,270 @@
+from dataclasses import dataclass
+from enum import Enum
+import errno
+from functools import cached_property
+import os
+import os.path
+import shutil
+
+import btrfsutil
+
+from bottles.backend.models.result import Result
+
+# TODO ask in the GUI, if a bottle should be created as subvolume.
+# TODO Add logging
+
+# Internal subvolumes created at initialization time:
+_internal_subvolumes = [
+    "cache",
+    ]
```

Comment:
I'm considering to get rid of this `_internal_subvolumes`. The idea is to save some disk space, when creating snapshots. However it significantly increases the complexity of the code. It makes it prone to have partially performed operations, who's proper handling is complex. I'm not sure, but it might have inspired @evertonstz to implement a semi transaction system, which seems overkill to me. Filesystems are no acid databases.

Reply:
I definitely need some feedback of a maintainer:

 * Shall I remove the internal subvolumes in order to keep the code simple? 
 * The idea of "intent files", to improve reliability of more broader filesystem changes against crashes or power outages, seems very good. Afaik nothing in this regard is implemented in bottles. In my opinion this should be implemented as an extension of the `Task` system. It would be out of scope of this PR.